src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -165,64 +258,12 @@
  * <pre>
  * public class D {
  *   private String prefix;
- *   public String modifyPrefix(String prefix) {
- *     prefix = "^" + prefix + "$" // no violation (modification of parameter)
- *     return prefix; // modified method parameter is returned from the method
- *   }
- * }
- * </pre>
- * <p>
- * Examples of how the check works if validateOnlyOverlapping option is set to false:
- * </p>
- * <pre>
- * public static class A {
- *   private int field1;
- *   private int field2;
  *
- *   public A(int field1) {
- *     field1 = field1; // violation: Reference to instance variable "field1" needs "this".
- *     field2 = 0; // violation: Reference to instance variable "field2" needs "this".
- *     String field2;
- *     field2 = "0"; // No violation. Local var allowed
- *   }
- *
- *   void foo3() {
- *     String field1 = "values";
- *     field1 = field1; // violation:  Reference to instance variable "field1" needs "this".
- *   }
- * }
- *
- * public static class B {
- *   private int field;
- *
- *   public A(int f) {
- *     field = f; // violation:  Reference to instance variable "field" needs "this".
- *   }
- *
- *   String addSuffixToField(String field) {
- *     return field += "suffix"; // violation: Reference to instance variable "field" needs "this".
- *   }
- * }
- *
- * // If the variable is locally defined, there won't be a violation provided the variable
- * // doesn't overlap.
- * class C {
- *   private String s1 = "foo1";
- *   String s2 = "foo2";
- *
- *   C() {
- *     s1 = "bar1"; // Violation. Reference to instance variable 's1' needs "this.".
- *     String s2;
- *     s2 = "bar2"; // No violation. Local var allowed.
- *     s2 += s2; // Violation. Overlapping. Reference to instance variable 's2' needs "this.".
+ *   public String modifyPrefix(String prefix) {
+ *     prefix = "^" + prefix + "$";  // no violation (modification of parameter)
+ *     return prefix;               // modified method parameter is returned from the method
```

Comment:
it is not clear is it violation or not

Reply:
Done, it should be easier to read and understand now.